infrastructure/src/main/java/org/corfudb/infrastructure/management/ClusterStateContext.java

```diff
@@ -0,0 +1,223 @@
+package org.corfudb.infrastructure.management;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import org.corfudb.protocols.wireprotocol.ClusterState;
+import org.corfudb.protocols.wireprotocol.NodeState;
+import org.corfudb.protocols.wireprotocol.SequencerMetrics;
+import org.corfudb.protocols.wireprotocol.SequencerMetrics.SequencerStatus;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.util.NodeLocator;
+
+/**
+ * Cluster State Context maintains shared context composing of cluster connectivity.
+ * Created by zlokhandwala on 11/2/18.
+ */
+@Slf4j
+public class ClusterStateContext {
+
+    //  Local copy of the local node's sequencer server metrics.
+    @Getter
+    private volatile SequencerMetrics localSequencerMetrics
+            = new SequencerMetrics(SequencerStatus.UNKNOWN);
+
+    // A view of peers' connectivity.
+    // Connectivity of any unresponsive nodes responding. Updated by HealingDetector.
+    private volatile Set<String> responsiveNodesState = Collections.emptySet();
+    // Connectivity of any responsive nodes not responding. Updated by FailureDetector.
+    private volatile Set<String> unresponsiveNodesState = Collections.emptySet();
+    // Cluster View compiled by aggregating latest peer views from heartbeat responses.
+    private final Map<String, NodeState> clusterView = new HashMap<>();
+
+    /**
+     * Compound object containing the last node's heartbeat seen and the corresponding local heartbeat at which it was
```

Comment:
Contains the heartbeat of a node, remote or local (with it's HeartbeatTimestamp) as observed by the local node at it's current local HeartbeatTimestamp.